kyaml/setters2/set_test.go

```diff
@@ -1099,7 +1099,7 @@ openAPI:
         setter:
           name: replicas
           value: "3"
-          isSet: true
+          isSet: "true"
```

Comment:
I think"true" and "false" values not being quoted is another yaml 1.1 backwards compat bug that this fixes so these tests do need to be updated.

I updated #4386 with [comment](https://github.com/kubernetes-sigs/kustomize/issues/4386#issuecomment-1029211685) showing where this fails.